addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/Pair.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.neeo;
+
+/**
+ * Simple pair structure of two types
+ *
+ * @param <L> the left generic type
+ * @param <R> the right generic type
+ *
+ * @author Tim Roberts - initial contribution
+ */
+public class Pair<L, R> {
+
+    /** The left */
+    private final L left;
+
+    /** The right */
+    private final R right;
+
+    /**
+     * Instantiates a new pair.
+     *
+     * @param left the left
+     * @param right the right
+     */
+    public Pair(L left, R right) {
+        this.left = left;
+        this.right = right;
+    }
+
+    /**
+     * Gets the left.
+     *
+     * @return the left
+     */
+    public L getLeft() {
+        return left;
+    }
+
+    /**
+     * Gets the right.
+     *
+     * @return the right
+     */
+    public R getRight() {
+        return right;
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see java.lang.Object#hashCode()
+     */
+    @Override
+    public int hashCode() {
+        return left.hashCode() ^ right.hashCode();
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see java.lang.Object#equals(java.lang.Object)
+     */
+    @SuppressWarnings("rawtypes")
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof Pair)) {
+            return false;
+        }
+        Pair pairo = (Pair) o;
+        return this.left.equals(pairo.getLeft()) && this.right.equals(pairo.getRight());
+    }
+
+}
```

Comment:
Please check all files for [end of line at the end of file](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)